openapi3/schema.go

```diff
@@ -1231,3 +1273,113 @@ func isSliceOfUniqueItems(xs []interface{}) bool {
 func unsupportedFormat(format string) error {
 	return fmt.Errorf("Unsupported 'format' value '%s'", format)
 }
+
+func invalidDefaultValue(s *Schema, err error) error {
+	return fmt.Errorf("default value error when validating schema %s (default: %v, error: %s)", "*name*", s.Default, err.Error())
+}
+
+func validateDefaultProperty(s *Schema, f func(s *Schema) error) error {
+	if s.Default == nil {
+		return nil
+	}
+	if err := f(s); err != nil {
+		return invalidDefaultValue(s, err)
+	}
+	return nil
+}
+
+func validateStringDefaultProperty(s *Schema, f func(s *Schema) error) error {
+	if s.Default == nil {
+		return nil
+	}
+	_, ok := s.Default.(string)
+	if !ok {
+		return invalidDefaultValue(s, errors.New("not a string"))
+	}
+	if err := f(s); err != nil {
+		return invalidDefaultValue(s, err)
+	}
+	return nil
+}
+
+func validateNumberDefaultProperty(s *Schema, f func(s *Schema) error) error {
+	if s.Default == nil {
+		return nil
+	}
+	_, ok := s.Default.(float64)
+	if !ok {
+		return invalidDefaultValue(s, errors.New("not a number"))
+	}
+	if err := f(s); err != nil {
+		return invalidDefaultValue(s, err)
+	}
+	return nil
+}
+
+func dfltStr(s *Schema) string { return fmt.Sprintf("%v", s.Default) }
+
+func validateBooleanDefault(s *Schema) (err error) {
+	return validateDefaultProperty(s, func(s *Schema) (err error) { _, err = strconv.ParseBool(dfltStr(s)); return err })
```

Comment:
This is just a style thing but long lines in this case hurt readability.  your lambda can have CR's in it and it will compile just fine and read much easier IMHO.

Reply:
agree - I initially wanted to make this really tight and self explanatory on a single line, but the end result is a little less readable than I'd like - will change